dask/dataframe/core.py

```diff
@@ -98,7 +98,13 @@
 
 no_default = "__no_default__"
 
-GROUP_KEYS_DEFAULT = None if PANDAS_GT_150 else True
+GROUP_KEYS_DEFAULT = None
```

Comment:
It looks like this line may not be needed 

```suggestion
```

Reply:
Initially this is how I had it, but I get a complaint from mypy:

```
mypy.....................................................................Failed
- hook id: mypy
- exit code: 1

dask/dataframe/core.py:104: error: Incompatible types in assignment (expression has type "None", variable has type "bool")  [assignment]
```

Reply:
I like your version, it's shorter too. I don't see type hints used much in our codebase, but since we support Python 3.8+, we could be using them more, right? What's the general agreement on using them? More is good, less is good?